### ISSUE TITLE
[pytest-bdd] def browser_type -- getattr(): attribute name must be string

### ISSUE THREAD

**Issue by sqandrew** [microsoft/playwright-pytest#54]
Hi there. Thanks a lot for the plugin!
Would like to try it with pytest-bdd, however got the following error:
```
    @pytest.fixture(scope="session")
    def browser_type(playwright: Playwright, browser_name: str) -> BrowserType:
>       return getattr(playwright, browser_name)
E       TypeError: getattr(): attribute name must be string
```

related step definition:
```
@given("test_playwright")
def test_playwright(page):
    pass
```

env:
```
Python: 3.9.0, 
Packages: {'pytest': '6.2.2', 'py': '1.10.0', 'pluggy': '0.13.1'}, 
Plugins:  Faker-6.5.0, progress-1.2.2, base-url-1.4.2, bdd-4.0.2, xdist-2.2.1, metadata-1.11.0, playwright-0.1.1, repeat-0.9.1, forked-1.3.0, allure-pytest-bdd-2.8.36, rerunfailures-10.0
^```

please advise
thanks!

**Comment by mxschmitt** [comment]
Are you using class-based tests like the `unittest.Testcase` class? If so, then its a duplicate of https://github.com/microsoft/playwright-pytest/issues/55

**Comment by sqandrew** [comment]
> Are you using class-based tests like the `unittest.Testcase` class? If so, then its a duplicate of #55

No, we don't.

Moreover, it works OK with adding to conftest.py browser_type fixture copy with specific browser name:
```
_conftest.py_

@pytest.fixture(scope="session")
def browser_type(playwright, browser_name: str):
    return getattr(playwright, 'chromium')
```

**Comment by sqandrew** [comment]
looks like it isn't compatible with pytest-bdd where a test consists of a set of steps (special fixtures).
it stops working when 'page' is the argument of a bdd step.

**Comment by mxschmitt** [comment]
Would it be possible that you create a small reproducible for me in form of a Git repository? I never have used pytest-bdd before and having it directly runable to see the error would be awesome. Thanks!

**Comment by sqandrew** [comment]
> Would it be possible that you create a small reproducible for me in form of a Git repository? I never have used pytest-bdd before and having it directly runable to see the error would be awesome. Thanks!

sure, please try https://github.com/sqandrew/pytestbdd-pw-test

**Comment by mxschmitt** [comment]
Thanks! I verified that the fix in #56 works for this case too. Probably pytest_bdd is using class based tests under the hood.

**Comment by sqandrew** [comment]
great news! waiting for release

**Comment by mxschmitt** [comment]
I released version 0.1.2 which should fix it.

**Comment by sqandrew** [comment]
it helped, thanks!
